File: jobfolio-backend/src/main/java/com/harshadcodes/jobfolio/entity/SavedJob.java
package com.harshadcodes.jobfolio.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "saved_jobs")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class SavedJob {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long userId;

    private String jobTitle;
    private String company;
    private String jobUrl;
    private String location;
    private String source;

    private LocalDateTime savedAt = LocalDateTime.now();
}

File: frontend/src/types/savedJob.ts
export interface SavedJob {
  id: number;
  jobTitle: string;
  company: string;
  jobUrl: string | null;
  location: string | null;
  source: string | null;
  savedAt: string;
}

export interface SaveJobPayload {
  jobTitle: string;
  company: string;
  jobUrl?: string;
  location?: string;
  source?: string;
}